### PR TITLE
core/validatorapi: support ssz encoded request bodies

### PR DESCRIPTION
**Pull request by corverroos**
Enable support for ssz encoding in `validatorapi`.

category: feature
ticket: #1537

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -143,6 +143,7 @@ linters:
     - interfacer
     - ireturn
     - lll # Think about adding this (max line length)
+    - maintidx
```
Comment (corverroos) [inline .golangci.yml:146]: new linter, similar to "complexity" linters that we disable.

Comment (xenowits) [reply .golangci.yml:146]: so the default is `20`, so ideally any function with maintainability index > 20 should raise linter errors

**Review by dB2510 — APPROVED** [review]
LGTM!
I suggest to test this locally in compose with teku VCs.

File: core/validatorapi/router.go [hunk 2]
```diff
@@ -947,7 +959,7 @@ func writeError(ctx context.Context, w http.ResponseWriter, endpoint string, err
 
 // unmarshal parses the JSON-encoded request body and stores the result
```
Comment (gsora) [inline core/validatorapi/router.go:960]:
```suggestion
// unmarshal parses body with the appropriate unmarshaler based on the contentType and stores the result
```

**Review by gsora — APPROVED** [review]

**Pull request merged**